app.js

```diff
@@ -1,117 +1,186 @@
+        return fn({
+            field: "password",
+            error: "Password must contain at least 1 number and 1 letter and be at least 4 symbols or more.",
+            status: 400,
+        });
+    if (!repeatPassword || password !== repeatPassword)
+        return fn({
+            field: "repeatPassword",
+            error: "Repeat password section validation (passwords should be the same)",
+            status: 400,
+        });
+    if (!firstName || firstName.length < 3)
+        return fn({
+            field: "firstName",
+            error: "First name must contain 3 symbols or more.",
+            status: 400,
+        });
+    if (!lastName || lastName.length < 3)
+        return fn({
+            field: "lastName",
+            error: "Last name must contain 3 symbols or more.",
+            status: 400,
+        });
+    if (!age || typeof age !== "number" || age === 0)
+        return fn({
+            field: "age",
+            error: "Age must be a number and can't be zero",
+            status: 400,
+        });
+
+    hash({ password }, function (err, pass, salt, hash) {
+        if (err) return fn(err);
 
-app.use(express.static("public"));
+        db.setUser({ ...userData, password: hash, salt })
+            .then(() => {
+                const { password, ...visibleUserData } = userData;
+                return fn(null, visibleUserData);
+            })
+            .catch((error) => {
+                return fn(error);
+            });
+    });
+}
 
-const filterProjects = (searchTerm = "") => {
-    const lowerCaseSearchTerm = searchTerm.toLowerCase();
-    const filteredProjects = defaultProjects.filter(
-        (project) =>
-            project.name.toLowerCase().includes(lowerCaseSearchTerm) ||
-            project.text.toLowerCase().includes(lowerCaseSearchTerm)
-    );
+app.post("/signup", function (req, res, next) {
+    if (!req.body) return res.sendStatus(400);
+    register(req.body, function (err, user) {
+        if (err) {
+            console.log(err);
+            res.status(400).send(err);
+        }
+        if (user) {
+            res.json({ data: user });
+        } else {
+            res.sendStatus(403);
+        }
+    });
+});
 
-    return filteredProjects;
-};
+app.post("/refresh-token", function (req, res, next) {
+    try {
+        if (!req.body) return res.sendStatus(400);
+        const token = req.body.refreshToken;
+        const { id, username } = jwt.verify(token, "Secretkey123");
+        res.status(200).send({
+            id,
+            username,
+            ...generateTokens({ id, username }),
+        });
+    } catch (error) {
+        console.log(error);
+        res.status(401).send(error);
+    }
+});
 
-app.get("/projects", (req, res) => {
+app.get("/projects", authMiddleware, (req, res) => {
     const searchTerm = req.query.search;
-
-    res.json({ data: filterProjects(searchTerm) });
+    if (searchTerm) {
```

Comment:
are you sure we can't avoid if else?

Reply:
try to pass an empty string as searchTerm or null (if it's easier for you) and don't create multiple requests with and without searchTerm